kong/endpoint_permission_service.go

```diff
@@ -0,0 +1,146 @@
+package kong
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// RBACEndpointPermissionservice handles RBACEndpointPermissions in Kong.
+type RBACEndpointPermissionservice service
+
+// Create creates a RBACEndpointPermission in Kong.
+func (s *RBACEndpointPermissionservice) Create(ctx context.Context,
+	ep *RBACEndpointPermission) (*RBACEndpointPermission, error) {
+
+	if ep == nil {
+		return nil, errors.New("cannot create a nil endpointpermission")
+	}
+	if ep.Role.ID == nil {
```

Comment:
You can have a nil pointer panic here.
Please do `ep.Role == nil || ep.Role.ID == nil`.
Please change this in all places.